src/jit/hwintrinsiccodegenxarch.cpp

```diff
@@ -0,0 +1,222 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+/*XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
+XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
+XX                                                                           XX
+XX               Intel hardware intrinsic Code Generator                     XX
+XX                                                                           XX
+XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
+XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
+*/
+#include "jitpch.h"
+#ifdef _MSC_VER
+#pragma hdrstop
+#endif
+
+#ifndef LEGACY_BACKEND // This file is ONLY used for the RyuJIT backend that uses the linear scan register allocator.
+#ifdef _TARGET_XARCH_
+
+#include "emit.h"
+#include "codegen.h"
+#include "sideeffects.h"
+#include "lower.h"
+#include "gcinfo.h"
+#include "gcinfoencoder.h"
+
+void CodeGen::genHWIntrinsic(GenTreeHWIntrinsic* node)
+{
+    NamedIntrinsic intrinsicID = node->gtHWIntrinsicId;
+    InstructionSet isa         = compiler->isaOfHWIntrinsic(intrinsicID);
+    switch (isa)
+    {
+        case InstructionSet_SSE:
+            genSSEIntrinsic(node);
+            break;
+        case InstructionSet_SSE2:
+            genSSE2Intrinsic(node);
+            break;
+        case InstructionSet_SSE3:
+            genSSE3Intrinsic(node);
+            break;
+        case InstructionSet_SSSE3:
+            genSSSE3Intrinsic(node);
+            break;
+        case InstructionSet_SSE41:
+            genSSE41Intrinsic(node);
+            break;
+        case InstructionSet_SSE42:
+            genSSE42Intrinsic(node);
+            break;
+        case InstructionSet_AVX:
+            genAVXIntrinsic(node);
+            break;
+        case InstructionSet_AVX2:
+            genAVX2Intrinsic(node);
+            break;
+        case InstructionSet_AES:
+            genAESIntrinsic(node);
+            break;
+        case InstructionSet_BMI1:
+            genBMI1Intrinsic(node);
+            break;
+        case InstructionSet_BMI2:
+            genBMI2Intrinsic(node);
+            break;
+        case InstructionSet_FMA:
+            genFMAIntrinsic(node);
+            break;
+        case InstructionSet_LZCNT:
+            genLZCNTIntrinsic(node);
+            break;
+        case InstructionSet_PCLMULQDQ:
+            genPCLMULQDQIntrinsic(node);
+            break;
+        case InstructionSet_POPCNT:
+            genPOPCNTIntrinsic(node);
+            break;
+        default:
+            unreached();
+            break;
+    }
+}
+
+void CodeGen::genSSEIntrinsic(GenTreeHWIntrinsic* node)
+{
+    // TODO - Implement SSE intrinsic code generation
```

Comment:
I think it would be good to have an NYI for these methods that are not yet used/implemented.

Reply:
Will do, thanks.